### PR TITLE
Lp 1753593 st1 sc1 aws volumes

### PR DESCRIPTION
**Pull request by AdrianGPrado**
## Please provide the following details to expedite Pull Request review:

Launch Pad[1753593](https://bugs.launchpad.net/juju/+bug/1753593)

----

## Description of change

AWS has changed its volume instances types.
- New `st1` magnetic volume type, with: big data, data warehouses, log processing... as use cases.
- New `sc1` magnetic volume type, with: cold data storage, few scans per day... as use cases.
- `magentic` type marked as depreciated, volume type ssd will be the new default option.

AWS volume `st1` requires a minimum of 500 GiB of storage, which at 0.045$/month is a total cost 22.5$/month. With this monthly cost, ssd option provides a more flexible approach, since ssd instances start at 1GiB at 0.10 $/month.

## QA steps

New type `st1`.

    juju create-storage-pool myhdd-pool ebs volume-type=st1
    juju deploy postgresql --storage pgdata=myssd-pool,32G

New type `sc1`, with alias `cold-storage`.

    juju create-storage-pool cold-storage ebs volume-type=cold-storage
    juju deploy postgresql --storage pgdata=myssd-pool,32G

Check result

    $ ./juju list-storage-pools 
    Name          Provider  Attrs
    cold-storage  ebs       volume-type=cold-storage
    ebs           ebs       
    ebs-ssd       ebs       volume-type=ssd
    loop          loop      
    myhdd-pool    ebs       volume-type=st1
    rootfs        rootfs    
    tmpfs         tmpfs     

## Documentation changes

PR [juju docs 2530](https://github.com/juju/docs/pull/2530)

## Bug reference

Launch Pad[1753593](https://bugs.launchpad.net/juju/+bug/1753593)

### CODE REVIEW

**Comment by babbageclunk** [comment]
You've explained why the change is needed in the description (which is great), but it would be good to also explain here why you've selected gp2 as the new default.

File: provider/ec2/ebs.go [hunk 1]
```diff
@@ -43,11 +43,14 @@ const (
 	EBS_Encrypted = "encrypted"
 
 	volumeTypeMagnetic        = "magnetic"         // standard
+	volumeTypeColdStorage     = "cold-storage"     // sc1
 	volumeTypeSSD             = "ssd"              // gp2
 	volumeTypeProvisionedIops = "provisioned-iops" // io1
 	volumeTypeStandard        = "standard"
```
Comment (wallyworld) [inline provider/ec2/ebs.go]: As discussed, let's split this list into 2 halves and add an alias for st1

File: provider/ec2/ebs.go [hunk 2]
```diff
@@ -110,6 +113,18 @@ const (
 	// volumes. We take the minimum of volumeSize*maxProvisionedIopsSizeRatio and
 	// maxProvisionedIops.
 	maxProvisionedIops = 20000
+
+	// minSt1VolumeSizeGiB is the minimum volume size for st1 volume instances.
+	minSt1VolumeSizeGiB = 500
+
+	// maxSt1VolumeSizeGiB is the minimum volume size for st1 volume instances.
```
Comment (wallyworld) [inline provider/ec2/ebs.go]: s/minimum/maximum

File: provider/ec2/ebs.go [hunk 3]
```diff
@@ -110,6 +113,18 @@ const (
...
+	maxSt1VolumeSizeGiB = 16 * 1024
+
+	// minSc1VolumeSizeGiB is the minimum volume size for sc1 volume instances.
+	minSc1VolumeSizeGiB = 500
+
+	// maxSc1VolumeSizeGiB is the minimum volume size for sc1 volume instances.
```
Comment (wallyworld) [inline provider/ec2/ebs.go]: ditto

File: provider/ec2/ebs.go [hunk 4]
```diff
@@ -160,7 +178,7 @@ var ebsConfigFields = schema.Fields{
 var ebsConfigChecker = schema.FieldMap(
 	ebsConfigFields,
 	schema.Defaults{
-		EBS_VolumeType: volumeTypeMagnetic,
+		EBS_VolumeType: volumeTypeGP2,
```
Comment (wallyworld) [inline provider/ec2/ebs.go]: The default value needs to be the alias

**Comment by AdrianGPrado** [comment]
$$merge$$

**Pull request merged**